### PR TITLE
Extract backup logic from device state 

### PR DESCRIPTION
**Pull request by telphan**
Continue device-state refactor work

### CODE REVIEW

File: src/api-binder.ts [hunk 1]
```diff
@@ -325,13 +311,15 @@ export class APIBinder {
...
 		});
 
-		return (await this.balenaApi
-			.post({ resource: 'device', body: device })
-			// TODO: Remove the `as number` when we fix the config typings
-			.timeout(conf.apiTimeout)) as Device;
+		return (
+			(await this.balenaApi
+				.post({ resource: 'device', body: device })
+				// TODO: Remove the `as number` when we fix the config typings
```
Comment (CameronDiver) [inline src/api-binder.ts]: We can remove this comment now it looks like

File: src/lib/migration.ts [hunk 2]
```diff
@@ -256,3 +264,87 @@ export async function normaliseLegacyDatabase(
...
+			path.join('mnt/data', constants.migrationBackupFile),
+		);
+		if (!exists) {
+			return;
+		}
+		log.info('Migration backup detected');
+
+		await deviceState.setTarget(targetState);
+
+		const appId = parseInt(_.keys(targetState.local?.apps)[0], 10);
```
Comment (CameronDiver) [inline src/lib/migration.ts]: Can you put a multi-app warning here? I add them to the code for places that definitely won't support multiple applications

File: src/lib/migration.ts [hunk 3]
```diff
@@ -256,3 +264,87 @@ export async function normaliseLegacyDatabase(
...
+
+		if (isNaN(appId)) {
+			throw new BackupError('No appId in target state');
+		}
+
+		const volumes = targetState.local?.apps?.[appId].volumes;
```
Comment (CameronDiver) [inline src/lib/migration.ts]: :heart_eyes:

File: src/api-binder.ts (hunk 1 above)
Comment (telphan) [reply src/api-binder.ts]: ack

File: src/lib/migration.ts (hunk 2 above)
Comment (telphan) [reply src/lib/migration.ts]: ack

File: src/lib/migration.ts (hunk 3 above)
Comment (telphan) [reply src/lib/migration.ts]: It is a beauty :D

**Comment by telphan** [comment]
Please DO NOT MERGE, I am still not sure how to test this.

**Comment by telphan** [comment]
@CameronDiver tested

**Pull request merged**